coderd/workspaces.go

```diff
@@ -952,6 +953,29 @@ func (api *API) putWorkspaceDormant(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// We don't need to notify the owner if they are the one making the request.
+	if req.Dormant && apiKey.UserID != workspace.OwnerID {
+		initiator, err := api.Database.GetUserByID(ctx, apiKey.UserID)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error fetching user.",
+				Detail:  err.Error(),
+			})
+			return
+		}
+		dormancy.NotifyWorkspaceDormant(
```

Comment:
Is this the only place this will be called? If so, I don't see the need for the helper method or the struct.

Reply:
Nops, we are going to call them in the lifecycle executor and the scheduler.